lib/jwt_sessions.rb

```diff
@@ -89,8 +87,19 @@ def algorithm
     @algorithm ||= DEFAULT_ALGORITHM
   end
 
+  def token_store=(args)
+    @token_store =
+      if args.is_a?(StoreAdapters::AbstractStoreAdapter)
+        args
+      elsif args.is_a?(Array)
+        StoreAdapters.build_by_name(args.first, args.last)
+      elsif args
+        StoreAdapters.build_by_name(args)
+      end
+  end
+
   def token_store
-    RedisTokenStore.instance(redis_url, token_prefix)
+    @token_store ||= StoreAdapters::RedisStoreAdapter.build
```

Comment:
Using Redis adapter by default along with making it an optional dependency is confusing.

If understand correctly, our goal is to make it possible to use the gem without Redis at all as long as making the upgrade for existent users (who use Redis) as smooth as possible. Right?

I see the following options.

1) Enforce users to specify `token_store` explicitly in the configuration:

```ruby
def token_store
  return @token_store if instance_variable_defined?(:@token_store)
  raise "JWTSessions.token_store is not specified. Please, configure the gem bla-bla-bla"
end
```

2) Check whether `redis` is could be loaded and fallback to in-memory store with warning message:

```ruby
def token_store
  return @token_store if instance_variable_defined?(:@token_store)
  
  begin
    require "redis"
    self.token_store = :redis
  rescue LoadError
    warn <<~MSG
      Warning! JWTSessions uses in-memory token store.
      Since you haven't configured JWTSessions.token_store, we tried to use Redis but couldn't load the 'redis' gem.
      
      Make sure that 'redis' gem is present in your Gemfile (or configure a different store).
    MSG
    self.token_store = :memory
  end
end
```

@tuwukee @DmitryTsepelev WDYT?

Reply:
I like the second option because it won't affect the existing users who use redis in their apps while the first option will force them to make an additional configuration